### PR TITLE
Refinement of fake package per API review

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Added an HTTP status code parameter to to methods that return a response.  For methods that already had such a paraemter, it was relocated to be the first parameter.
Removed Responder[T].SetHeader method, replacing the functionality with an http.Header field in SetResponseOptions.
Cleaned up some wording in the server API doc comments.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Needs more test coverage in the `server` package.

File: sdk/azcore/fake/fake.go [hunk 1]
```diff
@@ -43,507 +31,110 @@ type TokenCredential struct {
...
-}
+type AddNonTerminalResponseOptions = exported.AddNonTerminalResponseOptions
 
 // SetTerminalResponseOptions contains the optional values for PollerResponder[T].SetTerminalResponse.
-type SetTerminalResponseOptions struct {
-	// place holder for future optional values
-}
-
-/////////////////////////////////////////////////////////////////////////////////////////////////////////////
-// the following APIs are intended for use by fake servers
```
Comment (jhendrixMSFT) [inline sdk/azcore/fake/fake.go:186]: These APIs were moved into the `server` package.

File: sdk/azcore/fake/fake.go [hunk 2]
```diff
@@ -43,507 +31,110 @@ type TokenCredential struct {
...
-}
-
-// SetHeader sets the specified header key/value pairs to be returned.
-// Call multiple times to set multiple headers.
-func (r *Responder[T]) SetHeader(key, value string) {
-	if r.content.Header == nil {
-		r.content.Header = http.Header{}
-	}
-	r.content.Header.Set(key, value)
+func (r *Responder[T]) SetResponse(httpStatus int, resp T, o *SetResponseOptions) {
```
Comment (jhendrixMSFT) [inline sdk/azcore/fake/fake.go:54]: I have trepidations about the `httpStatus` parameter.  Given that `SetResponse()` is _always_ for the success case, we've now opened the possibility of the customer specifying the wrong HTTP status code, either the wrong 2xx status code or some other code that doesn't make sense for the success case.

Comment (jhendrixMSFT) [reply sdk/azcore/fake/fake.go:54]: Per discussion, we will leave this as-is.  To help push customers into the pit of success, we will codegen comments containing the list of successful HTTP status codes and return a descriptive error message when there's a mismatch.

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**